api/src/main/java/uk/gov/hmcts/payment/api/controllers/OrderController.java

```diff
@@ -0,0 +1,254 @@
+import uk.gov.hmcts.payment.api.v1.model.exceptions.*;
+
+import javax.validation.Valid;
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.function.Function;
+
+@RestController
+@Api(tags = {"Order"})
+@SwaggerDefinition(tags = {@Tag(name = "OrderController", description = "Order REST API")})
+public class OrderController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OrderController.class);
+    private static final String FAILED = "failed";
+
+    @Autowired
+    private OrderDomainService orderDomainService;
+
+    @Autowired
+    private IdempotencyKeysRepository idempotencyKeysRepository;
+
+    @Autowired
+    private Payment2Repository payment2Repository;
+
+    @Autowired
+    private CreditAccountDtoMapper creditAccountDtoMapper;
+
+    @ApiOperation(value = "Add Order with Fees", notes = "Add Order with Fees")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Order Created"),
+        @ApiResponse(code = 400, message = "Order Creation Failed"),
+        @ApiResponse(code = 401, message = "Credentials are required to access this resource"),
+        @ApiResponse(code = 403, message = "Forbidden-Access Denied"),
+        @ApiResponse(code = 422, message = "Invalid or missing attribute"),
+        @ApiResponse(code = 404, message = "No Service found for given CaseType"),
+        @ApiResponse(code = 504, message = "Unable to retrieve service information. Please try again later"),
+        @ApiResponse(code = 500, message = "Internal Server")
+    })
+    @PostMapping(value = "/order")
+    @Transactional
+    public ResponseEntity<?> create(@Valid @RequestBody OrderDto orderDto, @RequestHeader(required = false) MultiValueMap<String, String> headers) {
+        return new ResponseEntity<>(orderDomainService.create(orderDto, headers), HttpStatus.CREATED);
+    }
+
+    @ApiOperation(value = "Create credit account payment", notes = "Create credit account payment")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Payment created"),
+        @ApiResponse(code = 400, message = "Payment creation failed"),
+        @ApiResponse(code = 403, message = "Payment failed due to insufficient funds or the account being on hold"),
+        @ApiResponse(code = 404, message = "Account information could not be found"),
+        @ApiResponse(code = 504, message = "Unable to retrieve account information, please try again later"),
+        @ApiResponse(code = 422, message = "Invalid or missing attribute"),
+        @ApiResponse(code = 412, message = "The order has already been paid"),
+        @ApiResponse(code = 417, message = "The order amount should be equal to order balance")
+    })
+    @PostMapping(value = "/order/{order-reference}/credit-account-payment")
+    @ResponseBody
+    @Transactional
+    public ResponseEntity<OrderPaymentBo> createCreditAccountPayment(@RequestHeader(value = "idempotency_key") String idempotencyKey,
```

Comment:
It seems we are doing too much in controller layer. If we can move some of the logic to services layer, it makes things much more cleaner.

Reply:
Refactoring Done